src/manifold/src/constructors.cpp

```diff
@@ -322,28 +325,40 @@ Manifold Manifold::Extrude(const CrossSection& crossSection, float height,
  * calculated by the static Defaults.
  * @param revolveDegrees Number of degrees to revolve. Default is 360 degrees.
  */
-Manifold Manifold::Revolve(const CrossSection& crossSection,
-                           int circularSegments, float revolveDegrees) {
+Manifold Manifold::Revolve(const Polygons& crossSection, int circularSegments,
+                           float revolveDegrees) {
   ZoneScoped;
-  Polygons polygons = crossSection.ToPolygons();
 
-  if (polygons.size() == 0) {
-    return Invalid();
+  Polygons polygons;
+  float radius = 0;
+  for (const SimplePolygon& poly : crossSection) {
```

Comment:
It seems that our only non-trivial change is here. Do we have test cases that can cover it? While the logic is simple, it is easy to have bugs here.

Reply:
Indeed - yes this originally came in from an external contributor who put in test cases for it, so I think we're covered.